core/src/main/java/org/apache/iceberg/view/BaseView.java

```diff
@@ -97,4 +98,9 @@ public ReplaceViewVersion replaceVersion() {
   public UpdateLocation updateLocation() {
     return new SetViewLocation(ops);
   }
+
+  @Override
+  public UUID uuid() {
+    return UUID.fromString(ops.current().uuid());
```

Comment:
it should not go through an update. The purpose of `current()` is to return the currently loaded metadata, without checking for updates. If there's a catalog where `current()` does a refresh, then that's a bug

Reply:
I might be missing something here. but it seems `current()` can do the refresh [here](https://github.com/apache/iceberg/blob/cb20bdbea7b299b2948e908a7775ce90818e6a92/core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java#L75) .
unless some catalog has it's own implementation.

Reply:
that kind of (mandatory) refresh is expected, because it's a lazy refresh, that is typically done after a commit. This means that the underlying metadata changed after the commit, but we refresh only when accessing data after the commit happened. The same is being done in `BaseMetastoreTableOperations`